### PR TITLE
support  dubbo.registry.parameters.item3=value3 configuration properties

### PR DESCRIPTION
**Pull request by xiaoheng1**
## What is the purpose of the change

Support the following format configuration properties. such as:

dubbo.registry.parameters.item3=value3

see #8303 

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-common/src/main/java/org/apache/dubbo/config/AbstractConfig.java [hunk 1]
```diff
@@ -559,6 +560,13 @@ public void refresh() {
         postProcessRefresh();
     }
 
+    private void invokeSetParameters(Map<String, String> values) {
+        Map<String, String> map = invokeGetParameters(getClass(), this);
```
Comment (kylixs) [inline dubbo-common/src/main/java/org/apache/dubbo/config/AbstractConfig.java:564]: It is better to check argument `Map<String, String> values`, If it is null or empty, just return

File: dubbo-common/src/main/java/org/apache/dubbo/config/AbstractConfig.java [hunk 2]
```diff
@@ -540,10 +541,10 @@ public void refresh() {
...
                     String value = StringUtils.trim(subPropsConfiguration.getString(propertyName));
                     if (StringUtils.hasText(value)) {
-                        Map<String, String> map = invokeGetParameters(getClass(), this);
-                        map = map == null ? new HashMap<>() : map;
-                        map.putAll(convert(StringUtils.parseParameters(value), ""));
-                        invokeSetParameters(getClass(), this, map);
+                        invokeSetParameters(convert(StringUtils.parseParameters(value), ""));
+                    } else {
+                        // in this case, maybe parameters.item3=value3.
+                        invokeSetParameters(ConfigurationUtils.getSubProperties(subProperties, PARAMETERS));
```
Comment (kylixs) [inline dubbo-common/src/main/java/org/apache/dubbo/config/AbstractConfig.java:547]: Should we do `convert()` parameters or not? 
For compatibility, convert is needed, but additional parameters will be added, which leads to more url parameters, which is not good. This compatibility does not know what the specific problem is, it is best to deal with it when the parameters are read.

**Review by kylixs — CHANGES_REQUESTED** [review]

File: dubbo-common/src/main/java/org/apache/dubbo/config/AbstractConfig.java (hunk 1 above)
Comment (xiaoheng1) [reply dubbo-common/src/main/java/org/apache/dubbo/config/AbstractConfig.java:564]: ok, i will modify it.

**Review by kylixs — APPROVED** [review]

File: dubbo-common/src/main/java/org/apache/dubbo/config/AbstractConfig.java (hunk 2 above)
Comment (kylixs) [reply dubbo-common/src/main/java/org/apache/dubbo/config/AbstractConfig.java]: Some use a.b, some a-b, we will normalize parameter keys later.

**Pull request merged**